### PR TITLE
Fix the null exception when starting the proxy service

### LINKED ISSUE #7960: AuthenticationOAuth2 as Authentication Plugin in Pulsar Proxy
**Describe the bug**
When setting AuthenticationOAuth2 as brokerClientAuthenticationPlugin the proxy crashes on start-up with an uncaught Exception with message "null".

Relevant config:
brokerClientAuthenticationPlugin=org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2
brokerClientAuthenticationParameters={"grant_type": "client_credentials","issuerUrl": "https://example.com/your/auth/url/here","audience": "my_audience", "privateKey": "file:/path/to/keyfile/cc.json"}

The cc.json contains client_id and client_secret as described in the documentation.

**To Reproduce**
1. Set brokerClientAuthenticationPlugin and brokerClientAuthenticationParameters as described seen above.
2. Start the proxy.

**Expected behavior**
The proxy should be able to get a token from our OAuth provider and authenticate itself to the broker.

**Desktop (please complete the following information):**
 - OS: Pulsar is currently running in an kubernetes cluster. We also tried with the standalone version to no avail.

**Additional context**
The broker is defined with the  same config in terms of plugin and authenticationParameters and is able to get a token.

### PR DESCRIPTION
**Pull request by zymap**
---

Fixes #7960

*Motivation*

When enable the broker tls and enable broker client authentication with oauth2 plugin,
the proxy service will exit with an unexpected null exception.
Because there are some methods call the authentication before initializing the flow
which will cause the token client isn't initialized before using.

### CODE REVIEW

File: pulsar-proxy/pom.xml [hunk 1]
```diff
@@ -171,5 +171,12 @@
       <groupId>com.beust</groupId>
       <artifactId>jcommander</artifactId>
     </dependency>
+
+    <dependency>
+      <groupId>${project.groupId}</groupId>
+      <artifactId>bouncy-castle-bcfips</artifactId>
```
Comment (jiazhai) [inline pulsar-proxy/pom.xml:177]: @zymap Could this be changed into `bouncy-castle-bc` instead of bouncy-castle-bcfips? bc is used more widly.

**Review by jiazhai — APPROVED** [review]
overall lgtm. left a minor comments.

File: pulsar-proxy/pom.xml (hunk 1 above)
Comment (zymap) [reply pulsar-proxy/pom.xml]: Sure. Updated.

**Review by sijie — APPROVED** [review]

**Comment by EronWright** [comment]
@zymap I just noticed this fix, and do you recall why you fixed it in this way.  The true issue looks to me that the `Authentication` interface has a `start` method and a `getAuthData` method but that the `PulsarProxy` calls `getAuthData` before calling `start`.  The OAuth2 implementation assumes that `start` will be called first.  This PR avoids taking that assumption, but is it not a correct assumption to make?
Thanks

**Pull request merged**